services/media/service/src/domains/movies/handlers/publishing-movie-genres-processor.ts

```diff
@@ -1,40 +1,63 @@
 import {
+  MovieGenre,
   MovieGenresPublishedEvent,
   MovieGenresPublishedEventSchema,
   PublishServiceMessagingSettings,
 } from 'media-messages';
 import { all, Queryable, select } from 'zapatos/db';
-import { Config } from '../../../common';
+import { Config, DEFAULT_LOCALE_TAG } from '../../../common';
 import {
   buildPublishingId,
   EntityPublishingProcessor,
   SnapshotDataAggregator,
+  SnapshotValidationResult,
 } from '../../../publishing';
+import { getMovieGenreLocalizationsMetadata } from '../localization/get-movie-genre-localizations-metadata';
 
 const movieGenresDataAggregator: SnapshotDataAggregator = async (
   _entityId: number,
-  _authToken: string,
-  _config: Config,
+  authToken: string,
+  config: Config,
   queryable: Queryable,
 ) => {
   const genres = await select('movie_genres', all, {
     columns: ['id', 'sort_order', 'title'],
     order: { by: 'sort_order', direction: 'ASC' },
   }).run(queryable);
 
-  const snapshotJson: MovieGenresPublishedEvent = {
-    genres: genres.map((genre) => ({
-      title: genre.title,
+  const validations: SnapshotValidationResult[] = [];
+  const mappedGenres: MovieGenre[] = [];
+  for (const genre of genres) {
+    const { result: localizations, validation: localizationsValidation } =
+      await getMovieGenreLocalizationsMetadata(
```

Comment:
Could this be changed to send a single GQL query with like 20 genres at once to not have 20 separate requests?

Reply:
Not at the moment. Created a separate work item to optimize it in the future. #40874